### DISCUSSION TITLE
When would the dag file be generated?

### DISCUSSION THREAD

**Discussion by Tong-Chen** [nextflow-io/nextflow#4007] [General]
For nextflow before `22.04.5`, the dag file like `pipeline.dot` would be generated once the nextflow is started.

For nextflow in version `23.04`, the dag file like `pipeline.dot` would be generated after the nextflow is finished.

Is there any way to config to generate this dag file once started? Since I need this file to create the structure of the workflow.

Or is there another way to get the structure of the pipeline?

By the way, I could not find this file `.command.meta.json`  which was mentioned in <https://github.com/nextflow-io/nextflow/pull/3802> either. Is this file canceled? (I am using Nextflow version `23.05.0-edge`)

Thanks!

Chen Tong

**Comment by bentsherman** [comment]
Hi @Tong-Chen , the `-preview` CLI option was added between those versions, which you can use with the DAG report enabled to generate the DAG without running the pipeline. Maybe the DAG rendering was moved to the end of the pipeline execution as a side effect, I'm not sure.

As for the metadata file, that feature is still under development. If you see a PR that isn't "merged", that means it hasn't made it into the main codebase yet.

**Reply by Tong-Chen** [reply]
Thanks! I find the `-preview`  and could solve the problem in one ugly way. 

1. Run with `-preview` with dag.enablesd=true.
2. Then Run without `-preview` with dag.enablesd=false.
